Added SonarCloud integration (#172)
NLeSC/python-template: {{cookiecutter.project_slug}}/project_setup.md

-   [Relevant section in the
    guide](https://guide.esciencecenter.nl/#/best_practices/language_guides/python?id=coding-style-conventions)

## Continuous code quality

-   [Sonarcloud](https://sonarcloud.io/) is used to perform quality analysis and code coverage report on each push

> -   The GitHub organization and repository must be added Sonarcloud for analysis to work by going to

Reviewer: ```suggestion
-   The GitHub organization and repository must be added Sonarcloud for analysis to work
```

> -   The GitHub organization and repository must be added Sonarcloud for analysis to work by going to
>     [Sonarcloud](https://sonarcloud.io/projects/create), login with your GitHub account,
>     add organization or reuse existing and setup repository

Reviewer: ```suggestion
-   The GitHub organization and repository must be added Sonarcloud for analysis to work
    1. go to [Sonarcloud](https://sonarcloud.io/projects/create)
    2. login with your GitHub account
    3. add organization or reuse existing
    4. setup repository
```

> -   Analysis is run in [GH action workflow](.github/workflows/quality.yml)

Reviewer: ```suggestion
-   The analysis will be run by [GitHub Action workflow](.github/workflows/quality.yml)
```

-   [Relevant section in the
    guide](https://guide.esciencecenter.nl/#/best_practices/language_guides/python?id=coding-style-conventions)

## Continuous code quality

-   [Sonarcloud](https://sonarcloud.io/) is used to perform quality analysis and code coverage report on each push

> -   The GitHub organization and repository must be added Sonarcloud for analysis to work
>     1. go to [Sonarcloud](https://sonarcloud.io/projects/create)
>     2. login with your GitHub account
>     3. add organization or reuse existing
>     4. setup repository

Reviewer: ```suggestion
-   The GitHub organization and repository must be added Sonarcloud for analysis to work
    1. go to [Sonarcloud](https://sonarcloud.io/projects/create)
    2. login with your GitHub account
    3. add organization or reuse existing
    4. setup repository
    5. go to [new code definition administration page](https://sonarcloud.io/project/new_code?id={{ cookiecutter.github_organization }}_{{ cookiecutter.project_slug }}) and select `Number of days` option
```

-   [Relevant section in the
    guide](https://guide.esciencecenter.nl/#/best_practices/language_guides/python?id=coding-style-conventions)

## Continuous code quality

-   [Sonarcloud](https://sonarcloud.io/) is used to perform quality analysis and code coverage report on each push
-   The GitHub organization and repository must be added Sonarcloud for analysis to work
    1. go to [Sonarcloud](https://sonarcloud.io/projects/create)
    2. login with your GitHub account
    3. add organization or reuse existing
    4. setup repository
-   The analysis will be run by [GitHub Action workflow](.github/workflows/quality.yml)

> -   To run analysis a token must be created at [Sonarcloud account](https://sonarcloud.io/account/security/)
>     and token must be added as `SONAR_TOKEN` to [secrets on GitHub](https://github.com/{{ cookiecutter.github_organization }}/{{ cookiecutter.project_slug }}/settings/secrets/actions)

Reviewer: ```suggestion
-   To be able to run the analysis, a token must be created at [Sonarcloud account](https://sonarcloud.io/account/security/) and this token must be added as `SONAR_TOKEN` to [secrets on GitHub](https://github.com/{{ cookiecutter.github_organization }}/{{ cookiecutter.project_slug }}/settings/secrets/actions)
```
